pkg/ocm/multiclusterengine.go

```diff
@@ -0,0 +1,207 @@
+func NewMultiClusterEngineBuilder(apiClient *clients.Settings, name string) *MultiClusterEngineBuilder {
+	glog.V(100).Infof(
+		`Initializing new MultiClusterEngine structure with the following params: name: %s`, name)
+
+	builder := MultiClusterEngineBuilder{
+		apiClient: apiClient.Client,
+		Definition: &mceV1.MultiClusterEngine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+			Spec: mceV1.MultiClusterEngineSpec{},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the MultiClusterEngine is empty")
+
+		builder.errorMsg = "multiclusterengine 'name' cannot be empty"
+	}
+
+	return &builder
+}
+
+// WithOptions creates MultiClusterEngine with generic mutation options.
+func (builder *MultiClusterEngineBuilder) WithOptions(
+	options ...MultiClusterEngineAdditionalOptions) *MultiClusterEngineBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Setting MultiClusterEngine additional options")
+
+	for _, option := range options {
+		if option != nil {
+			builder, err := option(builder)
+
+			if err != nil {
+				glog.V(100).Infof("Error occurred in mutation function")
+
+				builder.errorMsg = err.Error()
+
+				return builder
+			}
+		}
+	}
+
+	return builder
+}
+
+// PullMultiClusterEngine loads an existing MultiClusterEngine into MultiClusterEngineBuilder struct.
+func PullMultiClusterEngine(apiClient *clients.Settings, name string) (*MultiClusterEngineBuilder, error) {
+	glog.V(100).Infof("Pulling existing MultiClusterEngine name: %s", name)
+
+	builder := MultiClusterEngineBuilder{
+		apiClient: apiClient.Client,
+		Definition: &mceV1.MultiClusterEngine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		builder.errorMsg = "multiclusterengine 'name' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("multiclusterengine object %s does not exist", name)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Get fetches the defined MultiClusterEngine from the cluster.
+func (builder *MultiClusterEngineBuilder) Get() (*mceV1.MultiClusterEngine, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting MultiClusterEngine %s", builder.Definition.Name)
+
+	MultiClusterEngine := &mceV1.MultiClusterEngine{}
+	err := builder.apiClient.Get(context.TODO(), runtimeClient.ObjectKey{
+		Name: builder.Definition.Name,
+	}, MultiClusterEngine)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return MultiClusterEngine, err
+}
+
+// Update modifies an existing MultiClusterEngine on the cluster.
+func (builder *MultiClusterEngineBuilder) Update() (*MultiClusterEngineBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating MultiClusterEngine %s", builder.Definition.Name)
+
+	err := builder.apiClient.Update(context.TODO(), builder.Definition)
+	builder.Object = builder.Definition
```

Comment:
This runs regardless of if the above operation is successful or not. if err != nil, you should return without updating builder.Object